chore: [#5] Updated configurations for Prettier

eldarcelik opened the pull request.

4zuko commented:
- Why are you using exact versions in `package.json`?
- Can you investigate the order of adding plugins to ESLint configuration? Should ESLint rules have higher priority than prettier rules?
- Can you also investigate which rule should be added so ESLint automatically adds a trailing comma? For example:

Instead of having:

```
const test = {
  a: 1,
  b: 'test',
  c: true
}
```

We should have:

```
const test = {
  a: 1,
  b: 'test',
  c: true,
}
```

eldarcelik commented:
> * Why are you using exact versions in `package.json`?
> * Can you investigate the order of adding plugins to ESLint configuration? Should ESLint rules have higher priority than prettier rules?
> * Can you also investigate which rule should be added so ESLint automatically adds a trailing comma? For example:
> 
> Instead of having:
> 
> ```js
> const test = {
>   a: 1,
>   b: 'test',
>   c: true
> }
> ```
> 
> We should have:
> 
> ```js
> const test = {
>   a: 1,
>   b: 'test',
>   c: true,
> }
> ```
> 
> Reason: If we add a new object attribute, the change will be shown in just one line. Otherwise, without trailing coma, we'll see the change in two lines while doing CR.

1. It was a mistake, I fixed it and also grouped dependencies based on their purpose.
2. The order of the plugins in the plugins array can influence how ESLint applies the rules. ESLint will apply its rules first and then let Prettier handle the formatting. Prettier is primarily focused on formatting, and you generally want it to have the last word in how your code looks. By placing it at the end, you ensure that it doesn't interfere with other ESLint rules.
3. Added that **`comma-dangle`** rule to automatically add a trailing comma. It is very good to have this rule when we come to the reviewing PRs!
